File: app/src/main/java/com/vaibhav/moviedetails/commons/LogUtils.java
package com.vaibhav.moviedetails.commons;

import android.util.Log;

import com.vaibhav.moviedetails.BuildConfig;

/**
 * @author Vaibhav Bhandula on 09/10/18.
 */
public class LogUtils {

    private static final boolean isLogEnabled = BuildConfig.DEBUG;

    public static void printLog(String tag, String message) {
        if (isLogEnabled) {
            Log.v(tag, message);
        }
    }

    public static void printStackTrace(Throwable throwable) {
        if (isLogEnabled) {
            throwable.printStackTrace();
        }
    }
}

File: app/src/main/java/com/vaibhav/moviedetails/commons/LogUtils.kt
package com.vaibhav.moviedetails.commons

import android.util.Log
import com.vaibhav.moviedetails.BuildConfig

/**
 * @author Vaibhav Bhandula on 08/04/19.
 */
object LogUtils {
    private val isLogEnabled = BuildConfig.DEBUG

    fun printLog(tag: String, message: String) {
        if (isLogEnabled) {
            Log.v(tag, message)
        }
    }

    fun printStackTrace(throwable: Throwable) {
        if (isLogEnabled) {
            throwable.printStackTrace()
        }
    }
}